crates/log-server/src/network.rs

```diff
@@ -217,6 +234,15 @@ impl RequestPump {
         trace!("All loglet workers have terminated");
     }
 
+    fn on_wait_for_tail(worker: &LogletWorkerHandle, msg: Incoming<WaitForTail>) {
+        if let Err(msg) = worker.enqueue_wait_for_tail(msg) {
+            // worker has crashed or shutdown in progress. Notify the sender and drop the message.
+            if let Err(e) = msg.to_rpc_response(TailUpdated::empty()).try_send() {
+                debug!(?e.source, peer = %e.original.peer(), "Failed to respond to WaitForTail message with status Disabled due to peer channel capacity being full");
```

Comment:
The debug message seems not aligned with the above comment and the response we are sending.

Reply:
The debug message is printed when we fail to send the response. The above comment is about why we are actually trying to respond now.

Reply:
True, makes sense 👍